app/lib/repository.rb

```diff
@@ -273,15 +274,15 @@ def self.get_all_permissions
         assignment.valid_groupings.each do |valid_grouping|
```

Comment:
We should not allow students who are in a hidden course to access repositories. Probably the best way to do this is to add an additional where statement in the `get_repo_auth_records` function to not consider assignments that belong to a hidden course.